flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -418,27 +424,31 @@ private static void assertRecoveredJobsAndDirtyJobResults(
                 "There should be no overlap between the recovered ExecutionPlans and the passed dirty JobResults based on their job ID.");
     }
 
-    private void startRecoveredJobs() {
+    private void startRecoveredJobs(boolean wrapIntoApplication) {
         for (ExecutionPlan recoveredJob : recoveredJobs) {
-            runRecoveredJob(recoveredJob);
+            runRecoveredJob(recoveredJob, wrapIntoApplication);
         }
         recoveredJobs.clear();
     }
 
-    private void runRecoveredJob(final ExecutionPlan recoveredJob) {
+    private void runRecoveredJob(
+            final ExecutionPlan recoveredJob, final boolean wrapIntoApplication) {
```

Comment:
Could you add some comments to explain the behavior around with regard to `wrapIntoApplication`?